### PR TITLE
Feat/clone public repos

### LINKED ISSUE #5365: Cannot list/clone/browse public GH repos
I can only see my own repos (or some _other_ repos I contributed to, I have no other clues...) in the GH pane.

Tangentially if I clone a public repo via query params (eg `?clone=concrete-utopia/hydrogen-november`) I cannot browse its branches on the pane (which is probably because of the same problem).

### PR DESCRIPTION
**Pull request by ruggi**
**Problem:**

It's not possible to clone public github repos that are not user repos.

**Fix:**

There are a lot of improvement we can do here iteratively, this PR is just the first step.
Now you can search for a repository (in the form `owner/repo`) in the repository field and the additional entry will include a `Search` button. Clicking it (or hitting enter in the input field) will try to find that repo on GH and, if available and pullable, continue with it in the rest of the integration flow.

Good ideas for incremental things:
1. disable the write blocks (commit and push, PR, etc) if the repo permissions don't include `push`
2. search repos by owner/org (omitting the `/repo` portion of the input string and adjusting the GH call done by the backend)
3. show the owner of the repo in the listing if it's not a `userRepository` but a `publicRepository`

**Notes:**
- A repo in the pane list is now marked as `searchable` directly from inside the additional row added for missing repos. The code in there is pretty messy and hard to follow and it would make a ton of sense to refactor it and clean it up, but I think it should be done separately.

**Manual Tests:**
I hereby swear that:

- [x] I opened a hydrogen project and it loaded
- [x] I could navigate to various routes in Preview mode

Fixes #5365

### CODE REVIEW

**Review by bkrmendy — APPROVED** [review]

File: editor/src/components/navigator/left-pane/github-pane/repository-listing.tsx [hunk 1]
```diff
@@ -107,27 +125,47 @@ const RepositoryRow = (props: RepositoryRowProps) => {
...
       <div>
         <Ellipsis style={{ maxWidth: 170 }}>{props.fullName}</Ellipsis>
         <span style={{ fontSize: 10, opacity: 0.5 }}>
-          {props.isPrivate ? 'private' : 'public'}
-          {props.updatedAt == null ? null : (
-            <>
-              {' '}
-              &middot; <TimeAgo date={props.updatedAt} />
-            </>
+          {when(
```
Comment (seanparsons) [inline editor/src/components/navigator/left-pane/github-pane/repository-listing.tsx:143]: You can flip the condition with `unless`.

Comment (ruggi) [reply editor/src/components/navigator/left-pane/github-pane/repository-listing.tsx]: https://github.com/concrete-utopia/utopia/pull/5370/commits/adf3413ca5bb934f22a7d70f442156c506eb1496

**Review by seanparsons — APPROVED** [review]
